### PR TITLE
Evo 1585 add duplicate post option

### LINKED ISSUE #1585: add Duplicate Post option to post action menu
Add a new menu item to the "three dot" menu in the PostHeader called "Duplicate".
This will open up the Create/Edit post modal (called the PostEditor component) and pre-populate it with all the data of the previous post, with the title changed to "Copy of [original post title truncated to fit into 80 characters with Copy Of at the front]", so that the user can edit the title, dates, content, etc. before they post it. 

One thing to test is whether this works ok for setting up the location object (make sure the new post appears on the map at the right location) and for the link preview (the link preview shows up in the post).

I will get you an icon for the Duplicate item soon!

old idea for reference, not doing this:
There's a few ways to accomplish this. All the data could be passed to the CreateModel and then down to the PostEditor via url parameters. We already have this option for lat/lng when a post is created by clicking on the map. But i don't think that would work for a very long description. Another option is to add a new "action" to the PosteEditor component. Right now we have 2 actions "create" and "edit". You would add a "copy" action. So then the URL you would open to duplicate a post with an id of 12345 would be /groups/xxxx/post/12345/copy. This would allow the PostEditor to pull the data from the store for the existing post. The main difference is that the Save button would change to say Duplicate, and when you click it it would call createPost instead of updatePost. That's probably the way to go.

### PR DESCRIPTION
**Pull request by KevinTriplett**
re-issuing PR with 4 organized commits to close #1585 
- add duplicate post option w/tests and translation
- add 'search' to params for stream param w/test (see comment below)
- fix three lint and test warnings
- enable Post button on duplicate post Modal entry

### CODE REVIEW

File: src/routes/AuthLayoutRouter/components/TopNav/NotificationsDropdown/NotificationsDropdown.test.js
Comment (KevinTriplett) [inline src/routes/AuthLayoutRouter/components/TopNav/NotificationsDropdown/NotificationsDropdown.test.js:1]: Lint complained that these variables were not used

File: src/routes/PostDetail/PostDetail.connector.test.js
Comment (KevinTriplett) [inline src/routes/PostDetail/PostDetail.connector.test.js:1]: Empty location caused console.log warning in getQueryParams.

**Review by thomasgwatson — APPROVED** [review]
lgtm, aside from the unanswered `q` or `search` matter.

**Comment by KevinTriplett** [comment]
> lgtm, aside from the unanswered `q` or `search` matter.

@thomasgwatson and @tibetsprague, I looked and per [line 79 of Stream.connector.js](https://github.com/Hylozoic/hylo-evo/blob/73a8e3f4f65d8b6ea0138bdc59c6459fb007096d/src/routes/Stream/Stream.connector.js#L78), 'q' is not used, it's 'search'

Also, Hylo search is unique to me: I can't do an OR search, it only accepts one search term at a time. Is that correct?

File: src/components/PostEditor/PostEditor.connector.js [hunk 1]
```diff
@@ -130,9 +130,9 @@ export const mergeProps = (stateProps, dispatchProps, ownProps) => {
   const goToPost = createPostAction => {
     const id = get('payload.data.createPost.id', createPostAction)
     // * The single letter params are used in the Stream and elsewhere
-    // and translate as follow: `s`(ort), `t`(ab), `q`(uery/search)
+    // and translate as follow: `s`(ort), `t`(ab), `q`(uery aliased as `search`)
     // The remaining whitelisted params are for the map view.
-    const querystringWhitelist = ['s', 't', 'q', 'zoom', 'center', 'lat', 'lng']
+    const querystringWhitelist = ['s', 't', 'q', 'search', 'zoom', 'center', 'lat', 'lng']
```
[the remark this answers is not in the document]
Comment (tibetsprague) [reply src/components/PostEditor/PostEditor.connector.js:135]: nice catch! i think we just never added search. however, this is working on production, i'm not losing the search no matter what i do when editing a post on the stream or map, so id have to dig deeper to see what is going on here. id say go ahead and add it

**Review by tibetsprague — COMMENTED** [review]
looks good, but i still need to add the icon, so i will do that then merge

**Comment by KevinTriplett** [comment]
> looks good, but i still need to add the icon, so i will do that then merge

I suggest we use the "copy" icon for duplicate and making a "link" icon for Copy Link. Because that's really what's happening.

File: src/components/PostEditor/PostEditor.connector.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (KevinTriplett) [reply src/components/PostEditor/PostEditor.connector.js:135]: yah, it's returned to whatever calls `mapStateToProps` but elsewhere, we're using `search` -- I'll see if I can find out how it's used (if it's used?) in the return from `mapStateToProps`

[the remark this answers is not in the document]
Comment (KevinTriplett) [reply src/components/PostEditor/PostEditor.connector.js:135]: @tibetsprague okay, yah, when editing a post, the search is retained. When creating a post, the search is reset. So this is a "bug" in that reverting this line to not include 'search' has a different behavior depending on whether it's an update or create.

**Comment by tibetsprague** [comment]
just realizing that you doing this development on a fork of the repo makes it harder for me to come in and collaborate on the branch 🤔 probably easier to just create branches in the main repo for now

**Comment by tibetsprague** [comment]
maybe you can create a PR to a branch in the central repo by the same name? then i can merge into that branch and do my work on that before merging to dev 🤔

**Comment by tibetsprague** [comment]
eh, for now ill just merge this to dev and add the icons there because im lazy

**Comment by tibetsprague** [comment]
well i found a bug after merging. if you change the post type of the duplicated post then it loses all the post content, because changing the post type changes the URL and gets rid of the fromPostId= parameter

**Comment by tibetsprague** [comment]
created ticket https://github.com/Hylozoic/hylo-evo/issues/1610

**Comment by tibetsprague** [comment]
also duplicating does not copy the images or attachments, which I think it should

**Comment by tibetsprague** [comment]
https://github.com/Hylozoic/hylo-evo/issues/1611

**Comment by KevinTriplett** [comment]
Want me to create a branch on the main repo and replicate the PR? I can do that.

Ah -- yah, I caught the missing attachments just today and wondered about it. Okay, thanks, I'll continue with the additional issues.

**Pull request merged**